File: src/Sequence/content/SqItems.java
package Sequence.content;

import Sequence.core.SeqElem;
import Sequence.world.meta.SqStat;
import arc.graphics.Color;
import mindustry.type.Item;
import mindustry.world.meta.Stat;
import mindustry.world.meta.StatValue;

public class SqItems {
    public static SqItem berylliumalAlloy,

    crystallizedBeryllium,

    encapsulatedImagineEnergy,

    grainBoundaryAlloy,

    pureSilicon,

    phaseCore,

    standardAlloyPlate,

    vectorizedChip;

    public static void load() {
        berylliumalAlloy = new SqItem("berylliumal-alloy") {{
            color = Color.valueOf("989aa4");
            cost = 0.95f;
            healthScaling = 1.3f;
        }};
        crystallizedBeryllium = new SqItem("crystallized-beryllium") {{
            color = Color.valueOf("d1eeee");
            healthScaling = 0.5f;
            cost = 0.65f;
        }};
        encapsulatedImagineEnergy = new EncapsulatedImagineEnergyItem("encapsulated-imagine-energy") {{
            color = Color.valueOf("bf92f9");
            radioactivity = 15;
            charge = 7;
            cost = 4.5f;
            healthScaling = 0.01f;
            explosiveness = 2;
        }};
        grainBoundaryAlloy = new SqItem("grain-boundary-alloy") {{
            color = Color.valueOf("aaaaaa");
            radioactivity = 0.3f;
            charge = 0.5f;
            cost = 1.8f;
        }};
        pureSilicon = new SqItem("pure-silicon") {{
            color = Color.valueOf("53565c");
            cost = 0.9f;
        }};
        phaseCore = new SqItem("phase-core") {{
            color = Color.valueOf("ffd59e");
            radioactivity = 2.5f;
            cost = 5;
            healthScaling = 0.1f;
        }};
        standardAlloyPlate = new SqItem("standard-alloy-plate") {{
            color = Color.valueOf("cbd97f");
            cost = 1.45f;
            healthScaling = 1.3f;
        }};
        vectorizedChip = new SqItem("vectorized-chip") {{
            color = Color.valueOf("4a4b53");
            cost = 1.2f;
            radioactivity = 0.85f;
            explosiveness = 0.05f;
        }};
    }

    public static class SqItem extends Item implements SeqElem {
        public int ord = -1;

        public SqItem(String name) {
            super(name);
        }

        @Override
        public int order() {
            return ord;
        }

        @Override
        public StatValue statValue() {
            return null;
        }
    }

    public static class EncapsulatedImagineEnergyItem extends SqItem {
        public EncapsulatedImagineEnergyItem(String name) {
            super(name);
        }

        @Override
        public void setStats() {
            super.setStats();
            stats.remove(Stat.explosiveness);
            stats.remove(Stat.radioactivity);
            stats.remove(Stat.charge);
            stats.remove(Stat.flammability);
            stats.addPercent(SqStat.informationAnnihilationFactor, 0.3f);
        }
    }
}

File: src/sequence/content/SqItems.kt
package sequence.content

import arc.graphics.Color
import mindustry.type.Item
import mindustry.world.meta.Stat
import sequence.core.SeqElem
import sequence.util.register
import sequence.world.meta.SqStat

object SqItems {
    lateinit var berylliumalAlloy: SqItem
    lateinit var crystallizedBeryllium: SqItem
    lateinit var encapsulatedImagineEnergy: SqItem
    lateinit var grainBoundaryAlloy: SqItem
    lateinit var pureSilicon: SqItem
    lateinit var phaseCore: SqItem
    lateinit var standardAlloyPlate: SqItem
    lateinit var vectorizedChip: SqItem
    lateinit var encapsulatedInverseEnergy: SqItem
    lateinit var particleContainer: SqItem

    fun load() {
        berylliumalAlloy = SqItem("berylliumal-alloy", Color.valueOf("989aa4")).register {
            cost = 0.95f
            healthScaling = 1.3f
        }
        crystallizedBeryllium = SqItem("crystallized-beryllium", Color.valueOf("d1eeee")).register {
            healthScaling = 0.5f
            cost = 0.65f
        }
        encapsulatedImagineEnergy =
            InformationAnnihilateItem("encapsulated-imagine-energy", Color.valueOf("bf92f9")).register {
                radioactivity = 15f
                charge = 7f
                cost = 4.5f
                healthScaling = 0.01f
                explosiveness = 2f
            }
        encapsulatedInverseEnergy =
            InformationAnnihilateItem("encapsulated-inverse-energy", Color.valueOf("f93a3a")).register {
                radioactivity = 22f
                cost = 0f
                healthScaling = -1f
                explosiveness = 20f
                informationAnnihilationFactor = -0.01f
            }
        grainBoundaryAlloy = SqItem("grain-boundary-alloy", Color.valueOf("aaaaaa")).register {
            radioactivity = 0.3f
            charge = 0.5f
            cost = 1.8f
        }
        pureSilicon = SqItem("pure-silicon", Color.valueOf("53565c")).register {
            cost = 0.9f
        }
        phaseCore = SqItem("phase-core", Color.valueOf("ffd59e")).register {
            radioactivity = 2.5f
            cost = 5f
            healthScaling = 0.1f
        }
        standardAlloyPlate = SqItem("standard-alloy-plate", Color.valueOf("cbd97f")).register {
            cost = 1.45f
            healthScaling = 1.3f
        }
        vectorizedChip = SqItem("vectorized-chip", Color.valueOf("4a4b53")).register {
            cost = 1.2f
            radioactivity = 0.85f
            explosiveness = 0.05f
        }
        particleContainer = SqItem("particle-container", Color.valueOf("ffd59e")).register {
            cost = 0.9f
        }
    }

    class SqItem @JvmOverloads constructor(name: String, color: Color = Color(Color.black)) : Item(name, color),
        SeqElem {
        var ord = -1
        override val order: Int
            get() = ord
    }

    class InformationAnnihilateItem @JvmOverloads constructor(name: String, color: Color = Color(Color.black)) :
        SqItem(name, color) {
        var informationAnnihilationFactor: Float = 0.3f

        override fun setStats() {
            super.setStats()
            stats.remove(Stat.explosiveness)
            stats.remove(Stat.radioactivity)
            stats.remove(Stat.charge)
            stats.remove(Stat.flammability)
            stats.addPercent(SqStat.informationAnnihilationFactor, informationAnnihilationFactor)
        }
    }
}
